mezo-org/mezod
# Fix random consensus failure

Refs: #276 

### Introduction

This PR fixes the random consensus failure outlined in #276.

### Changes

`CustomPrecompileGenesisAccounts` has been updated, sorting the accounts slice before returning it. This is required as we populate the slice by looping through the unordered `customPrecompiles` map, the ordering of the slice was inconsistent, resulting in the consensus failure.

#### TODO
- [x] Add unit tests for `CustomPrecompileGenesisAccounts`

### Testing


1. Init a fresh localnet chain: make localnet-bin-clean && make localnet-bin-init
2. Run 4 nodes using make localnet-bin-start
3. Wait a little bit. Ensure all nodes have reached a block height of at least 3
4. Repeat until you are happy the issue no longer randomly occurs.


---

### Author's checklist

- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`) - N/A

### Reviewer's checklist
